src/sources/util/mod.rs

```diff
@@ -1,5 +1,14 @@
 #[cfg(any(feature = "sources-http"))]
 mod body_decoding;
+#[cfg(any(
+    feature = "sources-kafka",
+    feature = "sources-socket",
+    feature = "sources-statsd",
+    feature = "sources-stdin",
+    feature = "sources-syslog",
+    feature = "sources-vector",
+))]
+pub mod decoding;
```

Comment:
I'm really not sure of the value of doing all this feature gating and nesting within `sources/util/`. If you asked me, I'd move all of this to a top-level `codecs/` and not worry about feature flags unless something brings in relatively heavy dependencies. That said, it's not something I feel strongly about.

Reply:
Just tried moving everything from `src/sources/util/decoding` to `lib/codec`. However, it's not easily possible since we depend on `src/internal_events` and other utils from `src/sources/util`

Reply:
Unfortunately, as long as we can't extract those pieces that we depend on to another library, the feature gating is needed to prevent clippy from reporting unused code

Reply:
Moved everything to top-level `codec` in ce1a6d35ac2d3fa8493700424e8e910d04285e65